Update nuxt, @nuxt/vue-app, @nuxt/types dependencies

khushi-kothari opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #3242 by @obulat

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
Updated nuxt dependency in ``frontend/package.json`` and updated the lock file, and commited the changes.

- [X] nuxt, @nuxt/vue-app, @nuxt/types: from 2.17.0 to ^2.17.2 [changes]

<img width="460" alt="Screenshot 2023-10-25 at 7 43 50 PM" src="https://github.com/WordPress/openverse/assets/62203565/396c1f6e-4f45-421c-bf2e-67c395fcc9ea">

[](url)

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->


## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [X] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [X] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat reviewed (changes requested):
Great start!

You will need to run `just node-install` at the root of the repository. This will update the lock file, [`pnpm-lock.yaml`](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml). After you commit this file, the linting should be fixed.

khushi-kothari commented:
> just node-install

After running ``just node-install``,  [pnpm-lock.yaml](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml) was not updated. Can you point out what is the issue here? 

<img width="796" alt="Screenshot 2023-10-25 at 9 34 19 PM" src="https://github.com/WordPress/openverse/assets/62203565/95f019ad-781d-4c06-be8e-19d4116da99d">

obulat commented:
> @obulat After running `just node-install`, [pnpm-lock.yaml](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml) was not updated. Can you point out what is the issue here?

I really don't understand why your lock file is updated, @khushi-kothari 😕 

When I ran it locally on your branch, I got the following:
```
 j node-install                                                                                                                                                   ✔  8s  18:06:56 
pnpm i
Scope: all 4 workspace projects
readPackage: typescript@^4.9.5 => typescript@^4.9.5 in dependencies of @openverse/eslint-plugin
frontend                                 |  WARN  deprecated mkdirp@0.5.1
frontend                                 |  WARN  deprecated ufo@0.7.9
frontend                                 |  WARN  deprecated async-cache@1.1.0
frontend                                 |  WARN  deprecated svgo@1.3.2
frontend                                 |  WARN  deprecated source-map-resolve@0.6.0
frontend                                 |  WARN  deprecated sourcemap-codec@1.4.8
frontend                                 |  WARN  deprecated stable@0.1.8
frontend                                 |  WARN  deprecated chokidar@2.1.8
frontend                                 |  WARN  deprecated fsevents@1.2.13
frontend                                 |  WARN  deprecated querystring@0.2.1
frontend                                 |  WARN  deprecated sane@4.1.0
frontend                                 |  WARN  deprecated urix@0.1.0
frontend                                 |  WARN  deprecated source-map-resolve@0.5.3
frontend                                 |  WARN  deprecated core-js@2.6.12
frontend                                 |  WARN  deprecated querystring@0.2.0
frontend                                 |  WARN  deprecated resolve-url@0.2.1
frontend                                 |  WARN  deprecated source-map-url@0.4.1
frontend                                 |  WARN  deprecated w3c-hr-time@1.0.2
frontend                                 |  WARN  deprecated @npmcli/move-file@1.1.2
frontend                                 |  WARN  deprecated @npmcli/fs@1.1.0
frontend                                 |  WARN  deprecated uuid@3.4.0
frontend                                 |  WARN  deprecated @babel/plugin-proposal-private-property-in-object@7.21.11
Packages: +258 -121
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------
Progress: resolved 2478, reused 2469, downloaded 0, added 1, done
node_modules/.pnpm/nuxt@2.17.2_463pqhi6eauw3mjqcyq4fuonrm/node_modules/nuxt: Running postinstall script, done in 578ms
. postinstall$ pnpm --filter ./packages/* run build
│ No projects matched the filters in ".../openverse"
└─ Done in 338ms
Done in 20.1s
just frontend/run i18n:en
pnpm run i18n:en

> @openverse/frontend@2.2.1 i18n:en .../openverse/frontend
> pnpm i18n:get-translations --en-only


> @openverse/frontend@2.2.1 i18n:get-translations .../openverse/frontend
> node src/locales/scripts/get-translations "--en-only"

Successfully saved English translation to en.json.
just frontend/run i18n:copy-test-locales
pnpm run i18n:copy-test-locales

> @openverse/frontend@2.2.1 i18n:copy-test-locales .../openverse/frontend
> cp -r test/locales/* src/locales/

.../openverse  nuxt *66 !1  git status                                                                                                                                                   ✔  22s  18:07:56 
On branch nuxt
Your branch is up to date with 'khushi-kothari/nuxt'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   pnpm-lock.yaml

no changes added to commit (use "git add" and/or "git commit -a")
```
_(I redacted the openverse file paths)_

I committed the lock file and pushed it to your branch. 

Other options to try are running `pnpm install --filter=frontend` or running `just install` from the root (`openverse/`), but I'm not sure they would work better.

khushi-kothari commented:
> > @obulat After running `just node-install`, [pnpm-lock.yaml](https://github.com/WordPress/openverse/blob/main/pnpm-lock.yaml) was not updated. Can you point out what is the issue here?
> 
> I really don't understand why your lock file is updated, @khushi-kothari 😕
> 
> When I ran it locally on your branch, I got the following:
> 
> ```
>  j node-install                                                                                                                                                   ✔  8s  18:06:56 
> pnpm i
> Scope: all 4 workspace projects
> readPackage: typescript@^4.9.5 => typescript@^4.9.5 in dependencies of @openverse/eslint-plugin
> frontend                                 |  WARN  deprecated mkdirp@0.5.1
> frontend                                 |  WARN  deprecated ufo@0.7.9
> frontend                                 |  WARN  deprecated async-cache@1.1.0
> frontend                                 |  WARN  deprecated svgo@1.3.2
> frontend                                 |  WARN  deprecated source-map-resolve@0.6.0
> frontend                                 |  WARN  deprecated sourcemap-codec@1.4.8
> frontend                                 |  WARN  deprecated stable@0.1.8
> frontend                                 |  WARN  deprecated chokidar@2.1.8
> frontend                                 |  WARN  deprecated fsevents@1.2.13
> frontend                                 |  WARN  deprecated querystring@0.2.1
> frontend                                 |  WARN  deprecated sane@4.1.0
> frontend                                 |  WARN  deprecated urix@0.1.0
> frontend                                 |  WARN  deprecated source-map-resolve@0.5.3
> frontend                                 |  WARN  deprecated core-js@2.6.12
> frontend                                 |  WARN  deprecated querystring@0.2.0
> frontend                                 |  WARN  deprecated resolve-url@0.2.1
> frontend                                 |  WARN  deprecated source-map-url@0.4.1
> frontend                                 |  WARN  deprecated w3c-hr-time@1.0.2
> frontend                                 |  WARN  deprecated @npmcli/move-file@1.1.2
> frontend                                 |  WARN  deprecated @npmcli/fs@1.1.0
> frontend                                 |  WARN  deprecated uuid@3.4.0
> frontend                                 |  WARN  deprecated @babel/plugin-proposal-private-property-in-object@7.21.11
> Packages: +258 -121
> +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------------------------------------------------------------------------
> Progress: resolved 2478, reused 2469, downloaded 0, added 1, done
> node_modules/.pnpm/nuxt@2.17.2_463pqhi6eauw3mjqcyq4fuonrm/node_modules/nuxt: Running postinstall script, done in 578ms
> . postinstall$ pnpm --filter ./packages/* run build
> │ No projects matched the filters in ".../openverse"
> └─ Done in 338ms
> Done in 20.1s
> just frontend/run i18n:en
> pnpm run i18n:en
> 
> > @openverse/frontend@2.2.1 i18n:en .../openverse/frontend
> > pnpm i18n:get-translations --en-only
> 
> 
> > @openverse/frontend@2.2.1 i18n:get-translations .../openverse/frontend
> > node src/locales/scripts/get-translations "--en-only"
> 
> Successfully saved English translation to en.json.
> just frontend/run i18n:copy-test-locales
> pnpm run i18n:copy-test-locales
> 
> > @openverse/frontend@2.2.1 i18n:copy-test-locales .../openverse/frontend
> > cp -r test/locales/* src/locales/
> 
> .../openverse  nuxt *66 !1  git status                                                                                                                                                   ✔  22s  18:07:56 
> On branch nuxt
> Your branch is up to date with 'khushi-kothari/nuxt'.
> 
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         modified:   pnpm-lock.yaml
> 
> no changes added to commit (use "git add" and/or "git commit -a")
> ```
> 
> _(I redacted the openverse file paths)_
> 
> I committed the lock file and pushed it to your branch.
> 
> Other options to try are running `pnpm install --filter=frontend` or running `just install` from the root (`openverse/`), but I'm not sure they would work better.

Thanks for adding lock file here, I will look into what is causing this issue so next updates can be committed smoothly.

obulat commented:
> Ok @obulat that is the next task and am updating vue dependencies. Why are some of the CI tests failing in the current PR?

The CI tests are failing because the `nuxt` updates have updated one of the `vue` packages, and now there is a version mismatch and Nuxt cannot be built. So, the Vue dependencies should actually be updated in this PR instead of a separate PR.

khushi-kothari commented:
Updated vue dependencies. Still 2 CI tests are failing.
Lock file updated this time and all changes added in the last commit that I added.
<img width="597" alt="Screenshot 2023-10-25 at 10 59 08 PM" src="https://github.com/WordPress/openverse/assets/62203565/b1677d55-ad89-421e-b093-b9d6d9dd0560">

obulat commented:
Weirdly, the lock file didn't fully match the updates in `package.json`, so the CI failed (you could see the messages about frozen lock file). I ran `just install` and committed the updated lock file. Let's see if the CI passes now :)

khushi-kothari commented:
Hey @obulat I have added all tasks as individual PRs. Do you need any more help from my side on this issue [#3242](https://github.com/WordPress/openverse/issues/3242)?

obulat commented:
> Hey @obulat I have added all tasks as individual PRs. Do you need any more help from my side on this issue [#3242](https://github.com/WordPress/openverse/issues/3242)?

Hi, @khushi-kothari ! Thank you for creating the individual PRs. Could you please update them so that they only update the dependencies listed in the corresponding PR description? 

For instance, [Update focus-trap dependency](https://github.com/WordPress/openverse/pull/3256) should only update `focus-trap`, but if you look at the files changed, you can see that it updates all of these packages:

- "@nuxtjs/i18n": "^7.3.1",    
- "@nuxtjs/sentry": "^7.5.0",    
- "@popperjs/core": "^2.11.8",
- "@tailwindcss/typography": "^0.5.10",    
- "clipboard": "^2.0.11",    "uuid": "^9.0.1",
- "vue": "^2.7.15",    
- "@types/uuid": "^9.0.6",    
- "autoprefixer": "^10.4.16",    
- "tailwindcss": "^3.3.3",    
- "vue-demi": "^0.14.6",    
- "vue-server-renderer": "^2.7.15",
- "vue-template-compiler": "^2.7.15",

khushi-kothari commented:
Hey @obulat I think as it is linting problem always there is some issue while tracking whitespace changes. I locally have the following change for some reason it is not tracked while commiting. 
<img width="268" alt="Screenshot 2023-10-26 at 8 03 55 PM" src="https://github.com/WordPress/openverse/assets/62203565/aac4317d-4f43-4608-b0d5-fa136bfa8cb1">
local ss:
<img width="926" alt="Screenshot 2023-10-26 at 7 27 50 PM" src="https://github.com/WordPress/openverse/assets/62203565/a36482d6-b3aa-4605-95ea-5e1d68ee2e97">

But while viewing it remotely, it is like this :
<img width="638" alt="Screenshot 2023-10-26 at 7 29 06 PM" src="https://github.com/WordPress/openverse/assets/62203565/5229ffbd-f01c-40fb-8e1e-5e5841bf92ed">

> I followed following steps in this PR :
> 1. Checked out the lock file from main to remove all unrelated changes.
> ``git checkout main -- pnpm-lock.yaml``
> 2. Ran just node-install to update the lock file.
> 3. Committed the updated lock file to the branch and pushed the changes.

Yet, linting error is thrown.

obulat commented:
> Hey @obulat I think as it is linting problem always there is some issue while tracking whitespace changes. I locally have the following change for some reason it is not tracked while commiting.

I am not sure what's happening with the whitespace changes and why are they not tracked... 
The lockfile changes, on the other hand, are not simply whitespace changes. In several PRs you can see that the Vue packages are still touched in the lock files, even though you reverted the Vue changes to the `package.json` file.
I could update the lock files for you if the process I described doesn't work for you locally:

> I followed following steps in this PR :
> 
> Checked out the lock file from main to remove all unrelated changes.
> git checkout main -- pnpm-lock.yaml
> Ran just node-install to update the lock file.
> Committed the updated lock file to the branch and pushed the changes.

khushi-kothari commented:
> > Hey @obulat I think as it is linting problem always there is some issue while tracking whitespace changes. I locally have the following change for some reason it is not tracked while commiting.
> 
> I am not sure what's happening with the whitespace changes and why are they not tracked... The lockfile changes, on the other hand, are not simply whitespace changes. In several PRs you can see that the Vue packages are still touched in the lock files, even though you reverted the Vue changes to the `package.json` file. I could update the lock files for you if the process I described doesn't work for you locally:
> 
> > I followed following steps in this PR :
> > Checked out the lock file from main to remove all unrelated changes.
> > git checkout main -- pnpm-lock.yaml
> > Ran just node-install to update the lock file.
> > Committed the updated lock file to the branch and pushed the changes.

Ya that will be very helpful as the process you described isn't working for me locally.

I hope my contributions were helpful. As my first experience to open sourcing, I could say I will continue doing it and will grow in the process. Thankyou for being the kindest mentor and guiding me through this issue very calmly.

obulat commented:
> Ya that will be very helpful @obulat as the process you described isn't working for me locally.
> 
> I hope my contributions were helpful. As my first experience to open sourcing, I could say I will continue doing it and will grow in the process. Thankyou for being the kindest mentor and guiding me through this issue very calmly.

Thank you for your contributions, @khushi-kothari, they are very helpful! Hope you can keep contributing with Openverse ❤️ 
It is often that dependency or local set up problems are more difficult than coding :)

obulat commented on frontend/package.json:
```diff
@@ -21,6 +21,7 @@
     "prod:playwright": "pnpm i18n:copy-test-locales && pnpm prod",
     "storybook": "nuxt storybook --port=54000",
     "storybook:build": "nuxt storybook build",
+    "storybook:build-for-docs": "NODE_OPTIONS=--openssl-legacy-provider nuxt storybook build",
```
This script is run in GitHub CI runner (not Docker), and without `openssl-legacy-provider` it fails.

dhruvkb reviewed (approved):
LGTM!
